src/main/java/me/karakelley/http/handlers/ApplicationAuthorizationFactory.java

```diff
@@ -0,0 +1,28 @@
+package me.karakelley.http.handlers;
+
+import me.karakelley.http.authorization.AlwaysAuthorized;
+import me.karakelley.http.authorization.BasicAuthorizer;
+import me.karakelley.http.filesystem.PublicDirectory;
+import me.karakelley.http.http.Authorization;
+import me.karakelley.http.server.Handler;
+
+import java.util.Map;
+
+public class ApplicationAuthorizationFactory {
+
+  private Authorization basicAuth;
+  private Authorization alwaysAuthorized;
+
+  public ApplicationAuthorizationFactory(Authorization basicAuth, Authorization alwaysAuthorized) {
```

Comment:
1. It's going to be hard to remember the order of these two arguments—"which one comes first? and what determines which one gets used again?" I don't have a great suggestion at the moment... 
2. The class name is better, but something still feels a little strange to me about it. Maybe just a personal style thing. You're creating a handler that combines an application with some form of auth protection/security. It's a subtle, "English-grammar" thing, but would `AuthorizedApplicationFactory` make more sense? (adjective before noun.) For some reason that name also feels to me a little more indicative of the fact that the logic in this class is specific to this application, whereas `ApplicationAuthorizationFactory` sounds like a potentially generic tool for you to build some kind of authorization object that someone/anyone might use for their own application. 🤔 does that make any sense?